lib/tasks/thumbnails.rake

```diff
@@ -0,0 +1,47 @@
+# coding: utf-8
+namespace :atla do
+  def update_thumbnail_reference work
+    solr_doc = SolrDocument.find(work.id)
+    # the path should actually point to the "downloads" folder.
+    return unless solr_doc['thumbnail_path_ss'].include? 'assets'
+
+    work.thumbnail = ActiveFedora::Base.find work.thumbnail_id
+    work.save
+  end
+
+  desc 'fix the thumbnail paths for all works in the database'
+  task update_all_thumbnail_paths: [:environment] do
+    Hyrax.config.curation_concerns.each do |cc|
+      cc.all.each do |work|
```

Comment:
I'm a bit confused on why you're querying for all the works (line 15) then requiring on line 4 for the SolrDocument.

Might you be able to write a single Solr query that looks for the models you want and to have a matching `thumbnail_path_ss`?

I know this is a one off task, but this is a potentially rather expensive set of query calls.

I'm thinking of https://github.com/samvera-labs/bulkrax/blob/302a085314af4dd7eeec734f641c4c67ac6b6a84/app/parsers/bulkrax/parser_export_record_set.rb#L155-L161. 

I assume that we could have a `q` parameter of `thumbnail_path_ss:*assets*` would query for thumbnail-paths that have the assets value in them.

Reply:
I agree that that is a better solution in theory, however I have never had much luck with solr wildcard searches. They don't seem reliable. I didn't have any failures locally so I tried this but searched for correct thumbnail fields instead, with unfortunate results. If you have recommendations on how to get this to work reliably Jeremy, then perhaps we could all pair tomorrow?

<img width="960" alt="Screen Shot 2023-04-13 at 18 47 04" src="https://user-images.githubusercontent.com/6759631/231920739-b74e742f-94db-4bb4-9482-23aa9f681648.png">

Reply:
@jeremyf +1 to braydon's suggestion above of a pairing session? my local testing didn't return the expected results either. 

![Screenshot 2023-04-14 at 9 29 18 AM](https://user-images.githubusercontent.com/29032869/232073411-42e09d16-c3c5-456c-9d20-ff958bd28228.png)

Reply:
we haven't found a solr solution, but @labradford did inform us that atla uses extended dismax.

since the first task is a one time operation, we're going to leave it alone. however, I decided to add a third task that can be run for a single work too if performance becomes an issue.